lib/puppet/provider/dsc_base_provider/dsc_base_provider.rb

```diff
@@ -560,6 +560,13 @@ def invoke_params(resource)
                                 end
     end
     params_block = interpolate_variables("$InvokeParams = #{format(params)}")
+    # HACK: Handle intentionally empty arrays - need to strongly type them because
+    # CIM instances do not do a consistent job of casting an empty array properly.
+    empty_array_parameters = resource[:parameters].select { |_k, v| v[:value].empty? }
+    empty_array_parameters.each do |name, properties|
+      param_block_name = name.to_s.gsub(/^dsc_/, '')
+      params_block = params_block.gsub("#{param_block_name} = @()", "#{param_block_name} = [#{properties[:mof_type]}]@()")
```

Comment:
JFC. I'm both glad you were able to figure this out, and saddened that you had to. I like it though.

Reply:
It's an extremely unclever hack but it _did_ work in this case.